### PR TITLE
Fix migration conditions and shared services invariant

### PR DESCRIPTION
**Pull request by tanya-borisova**
# PR for issue https://github.com/microsoft/AzureTRE/issues/2234

## What is being addressed

Fixing the condition in the migration script so that only duplicated and not deleted shared services are deployed.

As part of the fix, I have realised that he invariant for shared services is wrong. 
We need to have only one shared service at one time being active and having operations performed to, but we can't check for status as "deployed", as it can have multiple other statuses (eg "actionSucceeded", etc) and be perfectly valid. 

So I made it to check "not deleted", and "not failed to be deployed". If there is a shared service satisfying this criteria, new one cannot be deployed. 
This way we can still retry the deployment if the previous one fails.

### CODE REVIEW

**Review by ross-p-smith — APPROVED** [review]

**Review by damoodamoo — APPROVED** [review]
just the one snag but it looks good to moi

File: api_app/db/repositories/resources.py [hunk 1]
```diff
@@ -138,4 +138,4 @@ def get_timestamp(self) -> float:
 
 # Cosmos query consts
 IS_NOT_DELETED_CLAUSE = 'c.deploymentStatus != "deleted"'
-IS_DEPLOYED_CLAUSE = 'c.deploymentStatus = "deployed"'
+IS_OPERATING_SHARED_SERVICE = 'c.deploymentStatus != "deleted" && c.deploymentStatus != "deploymentFailed"'
```
[the remark this answers is not in the document]
Comment (tanya-borisova) [reply api_app/db/repositories/resources.py:141]: Quite right! I've changed so it uses the constants defined

**Pull request merged**